hi-ml/src/health_ml/runner.py

```diff
@@ -166,9 +168,16 @@ def run(self) -> Tuple[LightningContainer, AzureRunInfo]:
         :return: a tuple of the LightningContainer object and an AzureRunInfo containing all information about
             the present run (whether running in AzureML or not)
         """
-        # Usually, when we set logging to DEBUG, we want diagnostics about the model
-        # build itself, but not the tons of debug information that AzureML submissions create.
-        logging_to_stdout(logging.INFO if is_local_rank_zero() else "ERROR")
+        # Suppress the logging from all processes but the one for GPU 0 on each node, to make log files more readable
+        log_level = logging.INFO if is_local_rank_zero() else logging.ERROR
+        logging_to_stdout(log_level)
+        # When running in Azure, also output logging to a file. This can help in particular when jobs
+        # get preempted, but we don't get access to the logs from the previous incarnation of the job
+        if is_running_in_azure_ml():
+            rank = os.getenv(ENV_GLOBAL_RANK, "0")
+            timestamp = datetime.utcnow().strftime("%Y-%m-%dT%H%M%S")
+            filename = Path(OUTPUT_FOLDER) / f"logging_{timestamp}_rank{rank}.txt"
```

Comment:
Q: will each rank has its own log file? I think PL only allows rank 0 to log, should we add a subfolder for redirected std_logs? on 8 gpu nodes we'll have 8 files linving with the other outputs which might get messy